pkg/clusterclientstore/store.go

```diff
@@ -161,7 +162,8 @@ func (s *Store) syncCluster(name string) error {
 			return nil
 		}
 
-		return err
+		return shippererrors.NewKubeclientGetError("", name, err).
+			WithShipperKind("Cluster")
```

Comment:
Looks a bit foreign to the rest of the API

Reply:
I agree, I'm not a fan of it either, but that's the best I could come up with so far. Open to suggestions here.

For reference, I started with the following, but decided against it due to verbosity:

```go
return shippererrors.NewKubeclientGetError(shipper.SchemeGroupVersion.WithKind("Cluster"), "", name, err)
```